src/simulation/dynamics/stepperMotor/_UnitTest/test_stepperMotor.py

```diff
@@ -0,0 +1,213 @@
+
+    # Log the test module output message for data comparison
+    stepperMotorDataLog = StepperMotor.stepperMotorOutMsg.recorder()
+    unitTestSim.AddModelToTask(unitTaskName, stepperMotorDataLog)
+
+    # Initialize the simulation, set the sim run time, and execute the simulation
+    unitTestSim.InitializeSimulation()
+    actuateTime = stepTime * np.abs(stepsCommanded)  # [sec] Time for the motor to actuate to the desired angle
+    holdTime = 5  # [sec] Time the simulation will continue while holding the final angle
+    unitTestSim.ConfigureStopTime(macros.sec2nano(actuateTime + holdTime))
+    unitTestSim.ExecuteSimulation()
+
+    # Extract the logged data for plotting and data comparison
+    timespan = stepperMotorDataLog.times()
+    theta = (180 / np.pi) * stepperMotorDataLog.theta
+    thetaDot = (180 / np.pi) * stepperMotorDataLog.thetaDot
+    thetaDDot = (180 / np.pi) * stepperMotorDataLog.thetaDDot
+    motorStepCount = stepperMotorDataLog.stepCount
+    motorCommandedSteps = stepperMotorDataLog.stepsCommanded
+
+    # Only show plots if the motor actuates
+    if (stepsCommanded == 0):
+        show_plots = False
+
+    # Plot motor angle
+    plt.figure()
+    plt.clf()
+    plt.plot(timespan * macros.NANO2SEC, theta, label=r"$\theta$")
+    plt.title(r'Stepper Motor Angle $\theta_{\mathcal{F}/\mathcal{M}}$', fontsize=14)
+    plt.ylabel('(deg)', fontsize=14)
+    plt.xlabel('Time (s)', fontsize=14)
+    plt.legend(loc='upper right', prop={'size': 12})
+    plt.grid(True)
+
+    # Plot motor thetaDot
+    plt.figure()
+    plt.clf()
+    plt.plot(timespan * macros.NANO2SEC, thetaDot, label=r"$\dot{\theta}$")
+    plt.title(r'Stepper Motor Angle Rate $\dot{\theta}_{\mathcal{F}/\mathcal{M}}$', fontsize=14)
+    plt.ylabel('(deg/s)', fontsize=14)
+    plt.xlabel('Time (s)', fontsize=14)
+    plt.legend(loc='upper right', prop={'size': 12})
+    plt.grid(True)
+
+    # Plot motor thetaDDot
+    plt.figure()
+    plt.clf()
+    plt.plot(timespan * macros.NANO2SEC, thetaDDot, label=r"$\ddot{\theta}$")
+    plt.title(r'Stepper Motor Angular Acceleration $\ddot{\theta}_{\mathcal{F}/\mathcal{M}}$ ', fontsize=14)
+    plt.ylabel('(deg/s$^2$)', fontsize=14)
+    plt.xlabel('Time (s)', fontsize=14)
+    plt.legend(loc='upper right', prop={'size': 12})
+    plt.grid(True)
+
+    # Plot steps commanded and motor steps taken
+    plt.figure()
+    plt.clf()
+    plt.plot(timespan * macros.NANO2SEC, motorStepCount)
+    plt.plot(timespan * macros.NANO2SEC, motorCommandedSteps, '--', label='Commanded')
+    plt.title(r'Motor Step History', fontsize=14)
+    plt.ylabel('Steps', fontsize=14)
+    plt.xlabel('Time (s)', fontsize=14)
+    plt.legend(loc='upper right', prop={'size': 12})
+    plt.grid(True)
+
+    if show_plots:
+        plt.show()
+    plt.close("all")
+
+    # Check to ensure the initial angle converged to the reference angle
+    desiredMotorAngleTrue = initialMotorAngle + (stepsCommanded * stepAngle)
```

Comment:
We can remove these testMessages and fail count in favor of np.testing.assert

Reply:
ah I missed this comment, I will do that now.

Reply:
Done.